### PR TITLE
Homework5

### PR DESCRIPTION
**Pull request by kooldandy**

### CODE REVIEW

File: Task5/src/app.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import express from 'express';
+import * as bodyParser from 'body-parser';
+import { AppRouter } from './routes';
+import { AddressInfo } from 'net';
```
Comment (rchawla17) [inline Task5/src/app.ts:4]: Why did you use `AddressInfo` ?

File: Task5/src/app.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private initializeMiddlewares() {
+        this.app.use(bodyParser.urlencoded({ extended: true }));
+        this.app.use(bodyParser.json({
+            limit: '50mb',
+            verify(req: any, res, buf, encoding) {
+                req.rawBody = buf;
+            }
+        }));
+
+        //this.app.use(loggerRequest);
```
Comment (rchawla17) [inline Task5/src/app.ts:32]: Remove commented code

File: Task5/src/controller/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+export class UserController {
+    private userService: UserService;
+
+    constructor() {
+        this.userService = new UserService();
+    }
+
+    public getAllUsers = (req: Request, res: Response) => {
+        this.userService.getAllUsers()
+            .then((appusers: any) => res.status(200).send(appusers))
```
Comment (rchawla17) [inline Task5/src/controller/user.controller.ts:14]: use `constructResponse` & `constructErrorResponse` for all the APIs in user controller.

File: Task5/src/controller/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    public removeUserById = (req: Request, res: Response) => {
+        const id = parseInt(req.params.userId, 10);
+
+        this.userService.removeUserById(id)
+            .then((appusers: any) => res.status(200).send(appusers))
+            .catch((error: any) => res.status(400).send(error))
+    };
+
+    private constructResponse(result: any) {
```
Comment (rchawla17) [inline Task5/src/controller/user.controller.ts:50]: Remove this method.

File: Task5/src/model/usergroup.model.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import { Group, User } from './../model';
+
+export class UserGroup extends Model {
+    public userId!: number;
+    public groupId!: number;
+}
+
+UserGroup.init(
+    {
+        userId: {
```
Comment (rchawla17) [inline Task5/src/model/usergroup.model.ts:12]: add primary key in the table.

File: Task5/src/model/user.model.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "nodeuser",
+    sequelize: database,
+  }
+);
+
+// User.belongsToMany(Group, {
```
Comment (rchawla17) [inline Task5/src/model/user.model.ts:34]: remove commented code.

File: Task5/src/model/group.model.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            allowNull: false,
+        }
+    },
+    {
+        tableName: "nodegroup",
+        sequelize: database
+    }
+);
+
+// Group.belongsToMany(User, {
```
Comment (rchawla17) [inline Task5/src/model/group.model.ts:36]: remove commented code.

File: Task5/src/schema/index.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import { Request, NextFunction, Response } from 'express';
+import Joi from 'joi';
+
+export function createUserSchema(req: Request, res: Response, next: NextFunction) {
+    const userSchema = Joi.object({
+        username: Joi.string().min(3).max(30).required(),
+        email: Joi.string().email().min(8).max(30).required()
```
Comment (rchawla17) [inline Task5/src/schema/index.ts:7]: password validation ?

File: Task5/src/app.ts (hunk 1 above)
Comment (kooldandy) [reply Task5/src/app.ts:4]: will remove in the next commit

File: Task5/src/app.ts (hunk 2 above)
Comment (kooldandy) [reply Task5/src/app.ts:32]: will remove it in next commit

File: Task5/src/controller/user.controller.ts (hunk 3 above)
Comment (kooldandy) [reply Task5/src/controller/user.controller.ts:14]: will cover it in next commit

File: Task5/src/controller/user.controller.ts (hunk 4 above)
Comment (kooldandy) [reply Task5/src/controller/user.controller.ts:50]: will remove it in next commit

File: Task5/src/model/usergroup.model.ts (hunk 5 above)
Comment (kooldandy) [reply Task5/src/model/usergroup.model.ts:12]: will cover it in next commit

File: Task5/src/model/user.model.ts (hunk 6 above)
Comment (kooldandy) [reply Task5/src/model/user.model.ts:34]: will remove it in next commit

File: Task5/src/model/group.model.ts (hunk 7 above)
Comment (kooldandy) [reply Task5/src/model/group.model.ts:36]: will remove it in next commit

File: Task5/src/schema/index.ts (hunk 8 above)
Comment (kooldandy) [reply Task5/src/schema/index.ts:7]: will cover it in next commit

**Pull request merged**